packages/core/src/utils/paths.ts

```diff
@@ -119,26 +126,35 @@ export function makeRelative(
 }
 
 /**
- * Escapes spaces in a file path.
+ * Escapes special characters in a file path like macOS terminal does.
+ * Escapes: spaces, parentheses, brackets, braces, semicolons, ampersands, pipes,
+ * asterisks, question marks, dollar signs, backticks, quotes, hash, and other shell metacharacters.
  */
 export function escapePath(filePath: string): string {
   let result = '';
   for (let i = 0; i < filePath.length; i++) {
-    // Only escape spaces that are not already escaped.
-    if (filePath[i] === ' ' && (i === 0 || filePath[i - 1] !== '\\')) {
-      result += '\\ ';
+    const char = filePath[i];
+    const isAlreadyEscaped = i > 0 && filePath[i - 1] === '\\';
```

Comment:
don't you also need to make sure that filePath[i-1] is not also escaped (e.g. proceeding character is not a \
Expect this simplifies to there being an odd number of `\` proceeding the current characters.